libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/targets/js/KotlinJsDcePlugin.kt

```diff
@@ -44,17 +45,17 @@ class KotlinJsDcePlugin : Plugin<Project> {
     }
 
     private fun processCompilation(project: Project, kotlinCompilation: KotlinCompilation<*>) {
-        val kotlinTaskName = kotlinCompilation.compileKotlinTaskName
-        val kotlinTask = project.tasks.findByName(kotlinTaskName) as? Kotlin2JsCompile ?: return
+        val kotlinTask = kotlinCompilation.compileKotlinTaskProvider
         val dceTaskName = lowerCamelCaseName(
             DCE_TASK_PREFIX,
             kotlinCompilation.target.disambiguationClassifier,
             kotlinCompilation.name.takeIf { it != KotlinCompilation.MAIN_COMPILATION_NAME },
             if (kotlinCompilation.target is KotlinWithJavaTarget<*>) TASK_SUFFIX else MPP_TASK_SUFFIX
         )
-        val dceTask = project.tasks.create(dceTaskName, KotlinJsDce::class.java).also {
+
+        val dceTask = project.registerTask<KotlinJsDce>(dceTaskName) {
             it.dependsOn(kotlinTask)
-            project.tasks.findByName("build")!!.dependsOn(it)
+            project.tasks.getByName("build").dependsOn(it)
```

Comment:
You want to use `named` here: 
![image](https://user-images.githubusercontent.com/1832925/73293409-aff3cd80-41b8-11ea-9df3-ebb80075c2a8.png)

Reply:
Resolved in https://github.com/JetBrains/kotlin/pull/2996, I'm waiting for the review.